nflow-engine/src/main/java/io/nflow/engine/service/ArchiveService.java

```diff
@@ -40,34 +50,162 @@ public ArchiveService(ArchiveDao archiveDao) {
    * @param batchSize Number of workflow hierarchies to archive in a single transaction. Typical value is 1-20. This parameter
    * mostly affects on archiving performance.
    * @return Total number of archived workflows.
+   *
+   * @deprecated Use the {@link #cleanupWorkflows(ArchiveConfiguration)} that allows better configuration of the actions
    */
-  @SuppressFBWarnings(value = "BAS_BLOATED_ASSIGNMENT_SCOPE", justification = "periodicLogger is defined in correct scope")
+  @Deprecated
   public int archiveWorkflows(DateTime olderThan, int batchSize) {
-    Assert.notNull(olderThan, "olderThan must not be null");
-    Assert.isTrue(batchSize > 0, "batchSize must be greater than 0");
-    archiveDao.ensureValidArchiveTablesExist();
-    log.info("Archiving starting. Archiving passive workflows older than {}, in batches of {}.", olderThan, batchSize);
+    return cleanupWorkflows(new Builder()
+            .setArchiveWorkflowsOlderThan(new Duration(olderThan, Instant.now()))
```

Comment:
Converting DateTime to duration and then back to DateTime later for the query might change the actual value a bit - I would keep it as DateTime to give the caller full control about the limit.

Reply:
This is only a backward compatible api. My plan is to delete this method in the next major release.
I would think that the new way of specifying how old, in relative time, instances will be operated on will be an easier to configure/use api.

Reply:
Ok